cg/services/order_validation_service/workflows/tomte/validation/inter_field/rules.py

```diff
@@ -81,3 +83,10 @@ def validate_mothers_in_same_case_as_children(order: TomteOrder) -> list[MotherN
         case_errors = get_mother_case_errors(case)
         errors.extend(case_errors)
     return errors
+
+
+def validate_subject_ids_different_from_case_names(order: TomteOrder) -> list[CaseSampleError]:
```

Comment:
I think it is better to type the return with the exact type of error the function returns